src/main/java/uk/gov/hmcts/reform/blobrouter/reconciliation/service/SummaryReportCreator.java

```diff
@@ -0,0 +1,53 @@
+package uk.gov.hmcts.reform.blobrouter.reconciliation.service;
+
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.blobrouter.data.envelopes.Envelope;
+import uk.gov.hmcts.reform.blobrouter.reconciliation.report.SummaryReport;
+import uk.gov.hmcts.reform.blobrouter.reconciliation.report.SummaryReportItem;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static java.util.Collections.emptyList;
+
+@Component
+public class SummaryReportCreator {
+
+    public SummaryReport createSummaryReport(
+        List<Envelope> receivedEnvelopes,
+        List<uk.gov.hmcts.reform.blobrouter.reconciliation.model.in.Envelope> reportedEnvelopes
+    ) {
+        List<Envelope> processedEnvelopes =
+            receivedEnvelopes == null ? emptyList() : receivedEnvelopes;
+        List<uk.gov.hmcts.reform.blobrouter.reconciliation.model.in.Envelope> supplierEnvelopes
+            = reportedEnvelopes == null ? emptyList() : reportedEnvelopes;
+
+        int actualCount = processedEnvelopes.size();
+        int reportedCount = supplierEnvelopes.size();
+
+        List<SummaryReportItem> receivedButNotReported = processedEnvelopes
+            .stream()
+            .filter(e ->
+                supplierEnvelopes.stream().noneMatch(s -> isEqualFile(e, s)))
+            .map(e -> new SummaryReportItem(e.fileName, e.container))
+            .collect(Collectors.toList());
+
+        List<SummaryReportItem> reportedButNotProcessed = supplierEnvelopes
+            .stream()
+            .filter(s ->
+                processedEnvelopes.stream().noneMatch(e -> isEqualFile(e, s)))
+            .map(s -> new SummaryReportItem(s.zipFileName, s.container))
+            .collect(Collectors.toList());
```

Comment:
been looking into this and kept on arguing about 4 streams used here. how about applying views instead? `Sets.difference()` on both sets (well they are lists now but..) will give desired result without reading through those names + streams + filters + maps + collects can do that later in upstream or in `SummaryReport` constructor :shrug:

Reply:
might be better but with quick thought as these are different types both type needs to override equals or Comparator depending on tree type  or we can use intermediate type so convert all two objects to same type, or we can use interface. if these are the options I would prefer current code.??

Reply:
well. i don't mind. it's java world. i'm just exploring otions of XOR'ing 2 different lists/sets../arrays + seeing 4 streams it triggers such thought. if no good option - i don't mind

Reply:
As I've mentioned on one of previous comments, we could map to `SummaryReportItem` straight away.
Or use a `Pair`.

Reply:
used `SummaryReportItem` with Sets.difference